lib/replica_pools/connection_proxy.rb

```diff
@@ -103,7 +103,11 @@ def within_leader_block?
 
     def route_to(conn, method, *args, &block)
       raise ReplicaPools::LeaderDisabled.new if ReplicaPools.config.disable_leader && conn == leader
-      conn.retrieve_connection.send(method, *args, &block)
+      if args[3].is_a?(Hash) # we need to ** keyword arguments, sending a Hash in their place is no longer allowed in Ruby 3
+        conn.retrieve_connection.send(method, *args[0..2], **args[3], &block)
```

Comment:
Curious where the split is coming from here, i.e why `args[3]`. The overall change looks good.

Reply:
Yeah, so the reason is because there are 3 positional arguments and then the 4th argument in this case is a Hash which is meant to be used for keyword arguments.

Reply:
Ah right, for some reason, I was thinking `send` was some Connection adapter method rather than ruby's `send`. I was looking for [this](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html) which aligns with these args (`arel, name = nil, binds =nil`). 

However, looking at this now and the 2 optional arguments, do we have to check `args[1]` and `args[2]` as a hash?

Reply:
Or can we make use of argument forwarding on ruby 3.0?
```ruby
def route_to(conn, ...)
...
  conn.retrieve_connection.send(...)
...
end
```

Reply:
I wasn't aware of argument forwarding but definitely sounds like what we want. I'll check it out, thanks @tedPen!

Reply:
Let me know how it goes, I'm going to be looking more at the ruby 3.0 upgrade for rosie and looks like this change will be necessary so I'm happy to help out however I can.

Reply:
Ok so you can't split out any args from `...` right I don't think, like `route_to(conn, ...)`, you can only do everything, like `route_to(...)`. This might come in a later update though. For now I just broke out the keyword args in a more sane way.